Klipper_Files/Extra module/ercf.py

```diff
@@ -119,7 +119,7 @@ def handle_connect(self):
         if self.gear_stepper is None:
             raise config.error(
                 "Manual_stepper gear_stepper must be specified")
-        self.ref_step_dist=self.gear_stepper.rail.steppers[0].get_step_dist()
```

Comment:
I'm afraid this won't be backward compatible with the gear_stepper with endstop pin defined. 

Should the code incorporate the `hasattr` call to make the code backward compatbile with the legacy configuration?

Reply:
Actually it should work in both cases as it is, `manual_stepper.py` takes care of proper assignment to the steppers array in either case, see the code below:

```python
        if config.get('endstop_pin', None) is not None:
            self.can_home = True
            self.rail = stepper.PrinterRail(
                config, need_position_minmax=False, default_position_endstop=0.)
            self.steppers = self.rail.get_steppers()
        else:
            self.can_home = False
            self.rail = stepper.PrinterStepper(config)
            self.steppers = [self.rail]
```